back/app/services/side_fx_comment_service.rb

```diff
@@ -47,8 +47,7 @@ def after_mark_as_deleted(comment, user, reason_code, other_reason)
   def before_destroy(comment, user); end
 
   def after_destroy(frozen_comment, user)
-    # Refresh the count of project participants by clearing the cache
-    ParticipantsService.new.clear_project_participants_count_cache(frozen_comment.post.project) if frozen_comment.post.project
+    ParticipantsService.new.clear_project_participants_count_cache(frozen_comment.idea.project)
```

Comment:
Don't we still need the check for the related project `if frozen_comment.post.project`?

I'm wondering if, without this, we might run into issues if project is destroyed just before the comment is... but maybe this can just never happen (as related comments are destroyed first when a project is deleted?)

Reply:
I'm not sure if it's possible to delete comments anymore, I think they now get marked as deleted? Otherwise, indeed, a moderator could delete a project at the same time as someone deletes their comment, but I consider this scenario to be too unlikely to really deal with it.

In general, there are a lot of `if comment.idea.project` checks that were removed in this PR. This is because initiatives, which have no projects, are no longer supported by comments, and so I see this as a part of the cleanup. As you point out, there could still be edge cases where we might want to keep checking this, but I think they are too unlikely and that ideally we should just follow up on Sentry issues instead.

Reply:
Makes sense. Thanks for clarifying.